smif/model/sector_model.py

```diff
@@ -410,6 +412,31 @@ def add_interventions(self, intervention_list):
 
         self._sector_model.interventions = intervention_list
 
+    def add_initial_conditions(self, initial_conditions):
+        """Adds initial conditions (state) for a model
+        """
+        state_data = filter(
+            lambda d: len(d.data) > 0,
+            [self.intervention_state_from_data(datum) for datum in initial_conditions]
+        )
+        self._sector_model._initial_state = list(state_data)
```

Comment:
Could this be slightly more simply `state_data = [self.intervention_state_from_data(datum) for datum in initial_conditions if datum.data]`

Using the truthiness of non-empty arrays and an 'if' instead of filter/lambda?

Reply:
Have made that change.

I'm not absolutely clear on the difference between `initial conditions` and `initial system` as given in these methods and whether we need them both - one of them has moved from sos_model.

`add_initial_conditions` shoves a list of initial conditions into a class attribute
`create_initial_system` passes the list of initial conditions to the `intialise` method of the SectorModel class